targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/mbed_crc_api.c

```diff
@@ -0,0 +1,65 @@
+#include "crc_api.h"
+
+#include "drivers/fsl_crc.h"
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+static crc_bits_t width;
+
+bool hal_crc_is_supported(const crc_mbed_config_t* config)
+{
+  if (config == NULL)
+    return false;
+
+  if ((config->polynomial & 0x80008000) == 0)
```

Comment:
It look like this check is to return false in case when CRC size is different than CRC16 or CRC32 - only these sizes can be handled by K64F hardware. 
Please check the predefined polynomial values:
```
POLY_16BIT_CCITT = 0x1021
POLY_32BIT_ANSI  = 0x04C11DB7
````

In both cases `hal_crc_is_supported` function will return `false`. I still suggest to add CRC size field into `crc_mbed_config` structure.

Reply:
Hm, I somehow overlooked those, I suppose there isn't a reasonable way to handle this other than specifying the width so I'll add the config.